addon/components/frost-toggle-button.js

```diff
@@ -0,0 +1,106 @@
+import Ember from 'ember'
+const {computed, Component, ViewUtils} = Ember
+import layout from '../templates/components/frost-toggle-button'
+import FrostEventsProxy from '../mixins/frost-events-proxy'
+
+export default Component.extend(FrostEventsProxy, {
+
+  // ==========================================================================
+  // Properties
+  // ==========================================================================
+
+  layout: layout,
+
+  classNames: ['frost-toggle-button'],
+
+  attributeBindings: [
+    'toggled',
+    'disabled'
+  ],
+
+  classNameBindings: [
+    'disabled'
+  ],
+
+  size: 'medium',
+  toggled: false,
+
+  // ==========================================================================
+  // Functions
+  // ==========================================================================
+
+  init () {
+    this._super(...arguments)
+
+    const toggled = this.toggled
+    Ember.assert(`The value passed to ${this.toString()}{toggled} must be boolean`,
```

Comment:
Pull in PropTypes for this and set the types appropriately and you can avoid all of this (best practice)